presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -981,6 +982,19 @@ public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<
         return new HiveTableHandle(schemaName, tableName);
     }
 
+    private Set<ColumnStatisticType> getSupportedColumnStatisticsForTemporaryTable(Type type)
+    {
+        // Temporary table statistics are not committed to metastore, so no need to call metastore for supported
+        // column statistics, instead locally determine.
+        try {
+            return MetastoreUtil.getSupportedColumnStatistics(type);
+        }
+        // In case there are unsupported column types (native format), specify a default column statistics.
+        catch (IllegalArgumentException e) {
```

Comment:
This function will be very similar to `getSupportedColumnStatistics`, replace throw ing error with `return ImmutableSet.of(TOTAL_SIZE_IN_BYTES);`